nsinit/docker.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/codegangsta/cli"
+	"github.com/docker/libcontainer"
+	"github.com/docker/libcontainer/cgroups/systemd"
+	"github.com/docker/libcontainer/configs"
+)
+
+const (
+	dockerRoot    = "/var/run/docker/execdriver/native"
+	stateFilename = "state.json"
+)
+
+func loadDockerFactory(context *cli.Context) (libcontainer.Factory, error) {
+	cgm := libcontainer.Cgroupfs
+	if systemd.UseSystemd() {
+		cgm = libcontainer.SystemdCgroups
+	}
+	return libcontainer.New(dockerRoot, cgm)
+}
+
+func loadState(root string) (*libcontainer.State, error) {
+	f, err := os.Open(filepath.Join(root, stateFilename))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, err
+		}
+		return nil, err
```

Comment:
both returns here is the same. We can probably just have one `return nil, err` instead.